src/main/java/com/fasterxml/jackson/core/util/BufferRecycler.java

```diff
@@ -82,6 +82,8 @@ public class BufferRecycler
     // Note: changed from simple array in 2.10:
     protected final AtomicReferenceArray<char[]> _charBuffers;
 
+    private BufferRecyclerPool _pool;
```

Comment:
I am thinking that better place for pool reference would be in `IOContext`; could be made final, not referenced from `BufferRecycler`. Although maybe keeping non-final wherever makes sense to avoid double-release?

Reply:
Releasing the same `BufferRecycler` twice to the same pool will compromise the pool integrity and I implemented it in this way exactly to avoid that risk. I suggest to leave it as it is, but of course I'm open to discuss it. For now I will add a comment in the code to better clarify this.